psi4/driver/json_wrapper.py

```diff
@@ -32,27 +32,206 @@
+        Please see molssi-qc-schema.readthedocs.io/en/latest/spec_components.html for further details.
+
+    Notes
+    -----
+    !Warning! This function is experimental and likely to change in the future.
+    Please report any suggestions or uses of this function on github.com/MolSSI/QC_JSON_Schema.
+
+    Examples
+    --------
+
+    """
+
+    # Clean a few things
+    _clean_psi_environ(clean)
+
+    # This is currently a forced override
+    json_data["schema_version"] = "v0.1"
+
+    json_data["provenance"] = {"creator": "Psi4", "version": psi4.__version__, "routine": "psi4.json.run_json"}
+
+    # Build molecule
+    mol = core.Molecule.from_schema(json_data)
+    # print(json.dumps(json_data, indent=2))
+
+    # Set options
+    for k, v in json_data["keywords"].items():
+        core.set_global_option(k, v)
+
+    # Setup the computation
+    method = json_data["model"]["method"]
+    core.set_global_option("BASIS", json_data["model"]["basis"])
+    kwargs = {"return_wfn": True, "molecule": mol}
+
+    # Handle special properties case
+    if json_data["driver"] == "properties":
+        if "properties" in json_data["model"]:
+            kwargs["properties"] = [x.lower() for x in json_data["model"]["properties"]]
+
+            extra = set(kwargs["properties"]) - can_do_properties_
+            if len(extra):
+                raise KeyError("Did not understand property key %s." % kwargs["properties"])
+        else:
+            kwargs["properties"] = list(can_do_properties_)
+
+
+    # Actual driver run
+    val, wfn = methods_dict_[json_data["driver"]](method, **kwargs)
+
+    # Pull out a standard set of SCF properties
+    psi_props = psi4.core.get_variables()
+    json_data["psi4:qcvars"] = psi_props
+
+    # Handle the return result
+    if json_data["driver"] == "energy":
+        json_data["return_result"] = val
+    elif json_data["driver"] in ["gradient", "hessian"]:
+        json_data["return_result"] = val.np.ravel().tolist()
+    elif json_data["driver"] == "properties":
+        ret = {}
+        mtd = json_data["model"]["method"].upper()
+        if "dipole" in kwargs["properties"]:
+            ret["dipole"] = [psi_props[mtd + " DIPOLE " + x] for x in ["X", "Y", "Z"]]
+        if "quadrupole" in kwargs["properties"]:
+            ret["quadrupole"] = [psi_props[mtd + " QUADRUPOLE " + x] for x in ["XX", "XY", "XZ", "YY", "YZ", "ZZ"]]
+        if "mulliken_charges" in kwargs["properties"]:
+            ret["mulliken_charges"] = wfn.get_array("MULLIKEN_CHARGES").np.ravel().tolist()
+        if "lowdin_charges" in kwargs["properties"]:
+            ret["lowdin_charges"] = wfn.get_array("LOWDIN_CHARGES").np.ravel().tolist()
+        if "wiberg_lowdin_indices" in kwargs["properties"]:
+            ret["wiberg_lowdin_indices"] = wfn.get_array("WIBERG_LOWDIN_INDICES").np.ravel().tolist()
+        if "mayer_indices" in kwargs["properties"]:
+            ret["mayer_indices"] = wfn.get_array("MAYER_INDICES").np.ravel().tolist()
```

Comment:
Small misconception: The JSON parser is here to match the MolSSI QC JSON Schema which may not match Psi4's expected output. For `properties` we are in a bit uncharted waters here and are out to provide something that solves industry requests now rather than wait for the Schema to full catch up.

The two issues you bought up seems like a module consistency issue rather than a schema issue. If I recall correctly I patch this up to work everywhere as expected except for the cc* star modules which seemed to be much more involved.

NO Occupations stores a `std::vector<std::vector<std::tuple<double, int, int> >>` which I am not yet sure how to handle. I mostly changed the charges as the data is overridden if you request both types of charges which leads to poor behavior in the output. We can add a change for the NO Occupations if you wish, but it is much more likely they will change in the future.

Reply:
Even if downstream QC_JSON wants properties divorced from method (that is, store DIPOLE_MOMENT unadorned and assumed to be attached to the method MP2), think it'd help psi-side in compensating for wave-like routes through the code to label properties as precisely as possible at the earliest point of calculation, usually c-side?

Reply:
Only the explicit return field is unadorned without further annotation as this is the result that was requested. If we ever want to store all properties computed at multiple levels of theory, this will likely have to be fit in somewhere in the properties or a new field. It would be great to label everything, I just worry about the onus it puts on us and downstream users.

Reply:
If you could patch up the cc* that would be fantastic and get everyone on a consistent page.